test/plots/barPlotTests.ts

```diff
@@ -1 +1,783 @@
 ///<reference path="../testReference.ts" />
+
+describe("Plots", () => {
+  describe("Bar Plot", () => {
+    describe ("orientations", () => {
+      it("rejects invalid orientations", () => {
+        assert.throws(() => new Plottable.Plots.Bar("diagonal"), Error);
+      });
+
+      it("defaults to vertical", () => {
+        let defaultPlot = new Plottable.Plots.Bar<number, number>();
+        assert.strictEqual(defaultPlot.orientation(), "vertical", "default Plots.Bar() are vertical");
+      });
+
+      it("sets orientation on construction", () => {
+        let verticalPlot = new Plottable.Plots.Bar<number, number>("vertical");
+        assert.strictEqual(verticalPlot.orientation(), "vertical", "vertical Plots.Bar()");
+
+        let horizontalPlot = new Plottable.Plots.Bar<number, number>("horizontal");
+        assert.strictEqual(horizontalPlot.orientation(), "horizontal", "horizontal Plots.Bar()");
+      });
+    });
+
+    const orientations = [Plottable.Plots.Bar.ORIENTATION_VERTICAL, Plottable.Plots.Bar.ORIENTATION_HORIZONTAL];
+    orientations.forEach((orientation) => {
+      const isVertical = orientation === Plottable.Plots.Bar.ORIENTATION_VERTICAL;
+      const basePositionAttr = isVertical ? "x" : "y";
+      const baseSizeAttr = isVertical ? "width" : "height";
+      const valuePositionAttr = isVertical ? "y" : "x";
+      const valueSizeAttr = isVertical ? "height" : "width";
+
+      describe(`rendering when ${orientation}`, () => {
+        const data = [
+          { base: "A", value: 1 },
+          { base: "B", value: 0 },
+          { base: "C", value: -1 }
+        ];
```

Comment:
We don't necessarily need the `data` variable here since it's stored inside `dataset`